### PR TITLE
scheduler: fix write leader max zombie duration

### PR DESCRIPTION
**Pull request by lhy1024**
Signed-off-by: lhy1024 <admin@liudos.us>

### What problem does this PR solve?

Issue Number: Ref #4949

`maxZombieDur` is not correct when the first priority is query in hot write scheduler, query is reported by store heartbeat rather than region heartbeat, so the interval should be base store heartbeat rather than region heartbeat

It's the same reason with [`tikvCollector.GetLoads`](https://github.com/tikv/pd/blob/bca2afae417921c9e31fd2c208eea485d4d3597d/server/statistics/collector.go#L52)

### Check List

Tests

- Unit test

### Release note

```release-note
None.
```

### CODE REVIEW

**Review by HunDunDM — APPROVED** [review]

**Comment by nolouch** [comment]
plz describe why this changes.

**Comment by lhy1024** [comment]
> query is reported by store heartbeat rather than region heartbeat.

query is reported by store heartbeat rather than region heartbeat, so the interval should be base store heartbeat rather than region heartbeat

File: server/schedulers/hot_region.go [hunk 1]
```diff
@@ -664,6 +650,25 @@ func (bs *balanceSolver) tryAddPendingInfluence() bool {
 	return true
 }
 
+// Depending on the source of the statistics used, a different ZombieDuration will be used.
+// If the statistics are from the sum of Regions, there will be a longer ZombieDuration.
+func (bs *balanceSolver) calcMaxZombieDur() time.Duration {
+	switch bs.resourceTy {
+	case writeLeader:
+		if bs.firstPriority == statistics.QueryDim {
```
Comment (nolouch) [inline server/schedulers/hot_region.go:658]: add some comments.

**Review by nolouch — APPROVED** [review]

**Pull request merged**